app/controllers/chargeback_controller.rb

```diff
@@ -713,21 +713,14 @@ def get_tags_all(category)
 
   def get_cis_all
     @edit[:cb_assign][:cis] = {}
-    if @edit[:new][:cbshow_typ] == "enterprise"
-      e_id = MiqEnterprise.first
-      @edit[:cb_assign][:cis]["#{e_id.id}"] = "Enterprise"
-    elsif @edit[:new][:cbshow_typ] == "storage"
-      Storage.all.each do |s|
-        @edit[:cb_assign][:cis][s.id] = s.name
-      end
-    elsif @edit[:new][:cbshow_typ] == "ext_management_system"
-      ExtManagementSystem.all.each do |ms|
-        @edit[:cb_assign][:cis][ms.id] = ms.name
-      end
-    elsif @edit[:new][:cbshow_typ] == "ems_cluster"
-      EmsCluster.all.each do |cl|
-        @edit[:cb_assign][:cis][cl.id] = cl.name
+    classtype =
+      if @edit[:new][:cbshow_typ] == "enterprise"
+        MiqEnterprise
+      else
+        @edit[:new][:cbshow_typ].classify.constantize
```

Comment:
@martinpovolny Thanks!

@aljesusg Can you add a whitelist in the similar way I did #7251, specifically: https://github.com/ManageIQ/manageiq/pull/7251/files#diff-65512d4378cfa992fb57fbceaaf14c59R256

For example, 

Create a whitelist of the various values we had previously that we're now producing here:
`@edit[:new][:cbshow_typ].classify`

Basically: `SOME_WHITELIST_VARIABLE = %w(Storage  ExtManagementSystem EmsCluster)`
Then, only allow those types:
```ruby
klass = @edit[:new][:cbshow_typ].classify
raise "some error" unless SOME_WHITELIST_VARIABLE.include?(klass)
klass.constantize
```